index.js

```diff
@@ -51,50 +53,66 @@ module.exports = (robot, options) => {
     // Wait a random delay to more evenly distribute requests
     const delay = options.delay ? options.interval * Math.random() : 0
 
-    robot.log.debug({repository, delay, interval: options.interval}, `Scheduling interval`)
+    app.log.info(
+      { repository, delay, interval: options.interval },
+      `Scheduling interval`
+    )
 
     intervals[repository.id] = setTimeout(() => {
       const event = {
         name: 'schedule',
-        payload: {action: 'repository', installation, repository}
+        payload: { action: 'repository', installation, repository }
       }
 
       // Trigger events on this repository on an interval
-      intervals[repository.id] = setInterval(() => robot.receive(event), options.interval)
+      intervals[repository.id] = setInterval(
+        () => app.receive(event),
+        options.interval
+      )
 
       // Trigger the first event now
-      robot.receive(event)
+      app.receive(event)
     }, delay)
   }
 
   async function eachInstallation (callback) {
-    robot.log.trace('Fetching installations')
-    const github = await robot.auth()
+    app.log.trace('Fetching installations')
+    const github = await app.auth()
 
-    const req = github.apps.getInstallations({per_page: 100})
-    await github.paginate(req, res => {
-      (options.filter ? res.data.filter(inst => options.filter(inst)) : res.data)
-        .forEach(callback);
-    })
+    const installations = await github.paginate(
+      github.apps.listInstallations.endpoint.merge({ per_page: 100 })
+    )
+
+    const filteredInstallations = options.filter
+      ? installations.filter(inst => options.filter(inst))
+      : installations
+    return filteredInstallations.forEach(callback)
   }
 
   async function eachRepository (installation, callback) {
-    robot.log.trace({installation}, 'Fetching repositories for installation')
-    const github = await robot.auth(installation.id)
-
-    const req = github.apps.getInstallationRepositories({per_page: 100})
-    return github.paginate(req, res => {
-      const repos = res.data.repositories;
-      (options.filter ? repos.filter(repo => options.filter(installation, repo)) : repos)
-        .forEach(async repository => callback(repository, github));
-    })
+    app.log.trace({ installation }, 'Fetching repositories for installation')
+    const github = await app.auth(installation.id)
+
+    const repositories = (await github.paginate(
+      github.apps.listRepos.endpoint.merge({ per_page: 100 })
+    ))
+      .map(page => page.repositories)
+      .flat()
```

Comment:
@gr2m Would ❤️ your 💭 on this implementation. It's a bit odd since each page comes .back as an array with an embedded `repositories` array like this:

```json
{
  "total_count": 2,
  "repositories": [
    {
      "id": 1296269,
      "node_id": "MDEwOlJlcG9zaXRvcnkxMjk2MjY5",
      "name": "Hello-World",
      "full_name": "octocat/Hello-World"
    },
    {
      "id": 1296270,
      "node_id": "NDEwOlJlcG9zaXRvcnkxMjk2MjY5",
      "name": "Another-Repository",
      "full_name": "octocat/Another-Repository"
    }
  ]
}
```

So this puts all the repositories into a single flattened array.

Reply:
I thought the `.paginate` method would normalize responses like this, but I expected the key would always be called "items": https://github.com/octokit/rest.js/commit/8373934ed21eb69c8dbacc3e06e1dc1566a6a7d8

Ugh damn, now I’m not sure what to do, I’d argue this is a bug but changing it in Octokit might break people’s code :(

Reply:
Thoughts on feature flagging and deprecating in the next release? So like:

```
if (result.value.data.repositories && options.featureFlagHere) {
  result.value.data = result.value.data.repositories
}
```

Reply:
I don’t think it’s worth it. But I would release the change as a breaking change, it will be simple enough to migrate to. And I will probably continue to support v16.x for some time.

I made an issue at https://github.com/octokit/rest.js/issues/1283

Reply:
Sounds good. I'm going to leave a comment here in scheduler to fix this later then.

Reply:
Actually I think we don’t need a breaking change, we can return the array in `value.data` and add the 3 additional properties to the array :)